supervisor/jobs_api.go

```diff
@@ -142,13 +142,24 @@ func (self JobAPI) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 		re := regexp.MustCompile(`^/v1/job/([a-fA-F0-9-]+)/run`)
 		id := uuid.Parse(re.FindStringSubmatch(req.URL.Path)[1])
+		taskChan := make(chan string, 1)
 
 		self.Tasks <- &db.Task{
-			Op:      db.BackupOperation,
-			Owner:   params.Owner,
-			JobUUID: id,
+			Op:           db.BackupOperation,
+			Owner:        params.Owner,
+			JobUUID:      id,
+			TaskUUIDChan: taskChan,
+		}
+
+		taskUUID := <-taskChan
+
+		if taskUUID == "" {
+			w.WriteHeader(500)
+			JSONLiteral(w, fmt.Sprintf(`{"ok":"error"}`))
```

Comment:
Why is this `{"ok":"error"}`? shouldn't we return something more like `{"error":"here is the reason for the error"}`?